### DISCUSSION TITLE
Starting project using ROBO-GYM

### DISCUSSION THREAD

**Discussion by FeroMateo** [jr-robotics/robo-gym#43] [Q&A]
Hello everyone! I'm just starting in this world of ROS and Gazebo, I have some questions about how this framework works.
Context: I want to implement a Deep Learning algorithm to my own robot to avoid collisions in my house. 

1) I want train the robot on my main computer and then use that progress in a Raspberry Pi 4 inside the robot. Is it possible? Or this framework only works in parallel of Virtual simulations and Real simulations.

2) If that is possible, should I begin making my own SDF/URDF model of my robot, the house and then implement this frameworks?

Well thank you so much for reading this, 
I am looking forward to any response.

**Comment by matteolucchi** [comment]
Hi @FeroMateo ! It's great to hear about your project! 

> I want train the robot on my main computer and then use that progress in a Raspberry Pi 4 inside the robot. Is it possible? Or this framework only works in parallel of Virtual simulations and Real simulations.

Yes this is possible, but let me give you a couple of notes:
- Deep Reinforcement Learning is complex, it involves a lot of software components and tuning, so if you simply want to do navigation you are probably better off with some standard navigation algorithms, for instance using MoveIt and the Navigation Stack. Otherwise, if you are getting into this project to practice and learn about AI and specifically Deep Reinforcement Learning I think this is a good start. 
- To be able to transfer directly from training in simulation to deploying to the real robot you will need to have an accurate simulation of your robot

> If that is possible, should I begin making my own SDF/URDF model of my robot, the house and then implement this frameworks?

Yes, that is a very good starting point, and even if you decide to go with MoveIt you will still need this to integrate your robot in ROS. Furthermore, if you are planning on expanding the capabilities of your robot, ROS provides a lot of open source packages for different applications, so I'd recommend you to go in that direction. 

I hope this helps. 

Cheers,
Matteo
